### PR TITLE
Add __unicode__ for all core models

### PR DESCRIPTION
**Pull request by steve-gregory**
The `__unicode__` method makes it easier to determine information about the objects you are looking at in Django. A lot of core models were missing this method, in those instances I chose the most useful information to display per resource.

### CODE REVIEW

File: api/renderers.py [hunk 1]
```diff
@@ -25,10 +25,16 @@ class BrowsableAPIRenderer(renderers.BrowsableAPIRenderer):
...
-    By setting 'display_edit_forms' to False in the context,
-    you can display the API without forms.
+    By returning an empty rendered HTML form
+    you can display the API quickly, without
+    having to deal with "select" queryset slow-downs.
     """
 
+    def get_rendered_html_form(self, data, view, method, request):
+        if method in ['PUT','POST']:
+            return ""
```
Comment (steve-gregory) [inline api/renderers.py]: This single change resulted in (Easily) a 50x speed-up of the Browsable API.

JSON rendering was never affected.

**Comment by steve-gregory** [comment]
Self-:shipit:

**Comment by lenards** [comment]
:+1:

**Comment by erbriones** [comment]
If you decorate the class with the `@python_2_unicode_compatible` you can future proof and use`__str__` over `__unicode__`.
https://docs.djangoproject.com/en/1.9/ref/utils/#django.utils.encoding.python_2_unicode_compatible

**Pull request merged**